CodeEdit/Features/Git/Client/GitClient+Remote.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  GitClient+Remote.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 11/17/23.
+//
+
+import Foundation
+
+extension GitClient {
+    /// Gets all remotes
+    /// - Parameter name: Name for remote
+    /// - Parameter location: URL string for remote location
+    func getRemotes() async throws -> [GitRemote] {
+        let command = "remote -v"
+        let output = try await run(command)
+        let remotes = parseGitRemotes(from: output)
+
+        return remotes
+    }
+
+    /// Add existing remote to local git
+    /// - Parameter name: Name for remote
+    /// - Parameter location: URL string for remote location
+    func addRemote(name: String, location: String) async throws {
+        _ = try await run("remote add \(name) \(location)")
+    }
+
+    /// Remove remote from local git
+    /// - Parameter name: Name for remote to remove
+    func removeRemote(name: String) async throws {
+        _ = try await run("remote rm \(name)")
+    }
+}
+
+func parseGitRemotes(from output: String) -> [GitRemote] {
+    var remotes: [String: (fetch: String?, push: String?)] = [:]
+
+    output.split(separator: "\n").forEach { line in
+        let components = line.split { $0 == " " || $0 == "\t" }
+        guard components.count == 3 else { return }
```

Comment:
Lets log when this fails, so we can catch more edge cases and handle them

Reply:
How should I log it? (this is new territory for me)

Reply:
I needed some logging for my LSP so I used `import os.log` because it was used once in some other file. But there isn't much logging in this application in general it looks like. Maybe we can just leave a note here and we can talk about more comprehensive logging in a session. For now we can leave this as is.